src/basic/recurse-dir.c

```diff
@@ -84,6 +85,14 @@ int readdir_all(int dir_fd,
                 if (ignore_dirent(entry, flags))
                         continue;
 
+                if (FLAGS_SET(flags, RECURSE_DIR_ENSURE_TYPE)) {
+                        r = dirent_ensure_type(dir_fd, entry);
+                        if (r == -ENOENT)
```

Comment:
maybe add comment `/* dentry gone by now? no problem, let's just suppress it */`